hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestBlockOutputStreamWithFailures.java

```diff
@@ -266,6 +264,7 @@ void testWatchForCommitDatanodeFailure(boolean flushDelay, boolean enablePiggyba
       assertEquals(0, keyOutputStream.getStreamEntries().size());
       // Written the same data twice
       byte[] bytes = ArrayUtils.addAll(data1, data1);
+      cluster.restartHddsDatanode(pipeline.getNodes().get(0), true);
       validateData(keyName, bytes, client.getObjectStore(), VOLUME, BUCKET);
```

Comment:
Shouldn't we restart datanodes at the end, in `finally` block?  Restarting them before validation feels like altering the test.

Reply:
done